src/Editor/ImGui/ImGuizmo.cpp

```diff
@@ -0,0 +1,1719 @@
+            res.x = y * v.z - z * v.y;
+            res.y = z * v.x - x * v.z;
+            res.z = x * v.y - y * v.x;
+
+            x = res.x;
+            y = res.y;
+            z = res.z;
+            w = 0.f;
+        }
+        void Cross(const vec_t& v1, const vec_t& v2)
+        {
+            x = v1.y * v2.z - v1.z * v2.y;
+            y = v1.z * v2.x - v1.x * v2.z;
+            z = v1.x * v2.y - v1.y * v2.x;
+            w = 0.f;
+        }
+        float Dot(const vec_t &v) const
+        {
+            return (x * v.x) + (y * v.y) + (z * v.z) + (w * v.w);
+        }
+        float Dot3(const vec_t &v) const
+        {
+            return (x * v.x) + (y * v.y) + (z * v.z);
+        }
+
+        void Transform(const matrix_t& matrix);
+        void Transform(const vec_t & s, const matrix_t& matrix);
+
+        void TransformVector(const matrix_t& matrix);
+        void TransformPoint(const matrix_t& matrix);
+        void TransformVector(const vec_t& v, const matrix_t& matrix) { (*this) = v; this->TransformVector(matrix); }
+        void TransformPoint(const vec_t& v, const matrix_t& matrix) { (*this) = v; this->TransformPoint(matrix); }
+
+        float& operator [] (size_t index) { return ((float*)&x)[index]; }
+        const float& operator [] (size_t index) const { return ((float*)&x)[index]; }
+    };
+
+    vec_t makeVect(float _x, float _y, float _z = 0.f, float _w = 0.f) { vec_t res; res.x = _x; res.y = _y; res.z = _z; res.w = _w; return res; }
+    vec_t vec_t::operator * (float f) const { return makeVect(x * f, y * f, z * f, w *f); }
+    vec_t vec_t::operator - () const { return makeVect(-x, -y, -z, -w); }
+    vec_t vec_t::operator - (const vec_t& v) const { return makeVect(x - v.x, y - v.y, z - v.z, w - v.w); }
+    vec_t vec_t::operator + (const vec_t& v) const { return makeVect(x + v.x, y + v.y, z + v.z, w + v.w); }
+    vec_t vec_t::operator * (const vec_t& v) const { return makeVect(x * v.x, y * v.y, z * v.z, w * v.w); }
+    vec_t vec_t::Abs() const { return makeVect(fabsf(x), fabsf(y), fabsf(z)); }
+    ImVec2 operator+ (const ImVec2& a, const ImVec2& b) { return ImVec2(a.x + b.x, a.y + b.y); }
+
+    vec_t Normalized(const vec_t& v) { vec_t res; res = v; res.Normalize(); return res; }
+    vec_t Cross(const vec_t& v1, const vec_t& v2)
+    {
+        vec_t res;
+        res.x = v1.y * v2.z - v1.z * v2.y;
+        res.y = v1.z * v2.x - v1.x * v2.z;
+        res.z = v1.x * v2.y - v1.y * v2.x;
+        res.w = 0.f;
+        return res;
+    }
+
+    float Dot(const vec_t &v1, const vec_t &v2)
+    {
+        return (v1.x * v2.x) + (v1.y * v2.y) + (v1.z * v2.z);
+    }
+
+    vec_t BuildPlan(const vec_t & p_point1, const vec_t & p_normal)
+    {
+        vec_t normal, res;
+        normal.Normalize(p_normal);
+        res.w = normal.Dot(p_point1);
+        res.x = normal.x;
+        res.y = normal.y;
+        res.z = normal.z;
+        return res;
+    }
+
+    struct matrix_t
+    {
+    public:
+
+        union
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') keyword 'union' not followed by a single space [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT003)